File: ModPractice/src/main/java/info/nukoneko/minecraft/modpractice/ModMain.java
package info.nukoneko.minecraft.modpractice;

import net.minecraft.init.Blocks;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.event.FMLInitializationEvent;

@Mod(modid = ModMain.MOD_ID, version = ModMain.VERSION)
public class ModMain {
    static final String MOD_ID = "modpractice";
    static final String VERSION = "0.0.1";

    @Mod.EventHandler
    public void init(FMLInitializationEvent event) {
        System.out.println("DIRT BLOCK >> " + Blocks.DIRT.getUnlocalizedName());
    }
}

File: modPractice/src/main/kotlin/ModMain.kt
import net.minecraft.block.Blocks
import net.minecraftforge.common.MinecraftForge
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.event.lifecycle.FMLCommonSetupEvent
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext

@Suppress("UNUSED_PARAMETER")
@Mod("mod_practice") // mods.toml と一致する必要がある
class ModMain {
    init {
        FMLJavaModLoadingContext.get().modEventBus.addListener<FMLCommonSetupEvent> {
            setup(it)
        }
        MinecraftForge.EVENT_BUS.register(this)
    }

    private fun setup(event: FMLCommonSetupEvent) {
        println("DIRT BLOCK >> ${Blocks.DIRT.name}")
    }
}
